test/unit/driver/test_digitalocean.py

```diff
@@ -0,0 +1,230 @@
+#  Copyright (c) 2015-2018 Cisco Systems, Inc.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to
+#  deal in the Software without restriction, including without limitation the
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
+import os
+
+import pytest
+
+from molecule import config
+from molecule.driver import digitalocean
+
+
+@pytest.fixture
+def _instance(patched_config_validate, config_instance):
+    return digitalocean.DigitalOcean(config_instance)
+
+
+def test_config_private_member(_instance):
+    assert isinstance(_instance._config, config.Config)
+
+
+def test_testinfra_options_property(_instance):
+    assert {
+        'connection': 'ansible',
+        'ansible-inventory': _instance._config.provisioner.inventory_file
+    } == _instance.testinfra_options
+
+
+def test_name_property(_instance):
+    assert 'digitalocean' == _instance.name
+
+
+def test_options_property(_instance):
+    x = {'managed': True}
+
+    assert x == _instance.options
+
+
+def test_login_cmd_template_property(_instance):
+    x = ('ssh {address} -l {user} -p {port} -i {identity_file} '
+         '-o UserKnownHostsFile=/dev/null '
+         '-o ControlMaster=auto '
+         '-o ControlPersist=60s '
+         '-o IdentitiesOnly=yes '
+         '-o StrictHostKeyChecking=no')
+
+    assert x == _instance.login_cmd_template
+
+
+def test_safe_files_property(_instance):
+    x = [
+        os.path.join(_instance._config.scenario.ephemeral_directory,
+                     'instance_config.yml'),
+    ]
+
+    assert x == _instance.safe_files
+
+
+def test_default_safe_files_property(_instance):
+    x = [
+        os.path.join(_instance._config.scenario.ephemeral_directory,
+                     'instance_config.yml'),
+    ]
+
+    assert x == _instance.default_safe_files
+
+
+def test_delegated_property(_instance):
+    assert not _instance.delegated
+
+
+def test_managed_property(_instance):
+    assert _instance.managed
+
+
+def test_default_ssh_connection_options_property(_instance):
+    x = [
```

Comment:
Optional BUT it would be great to have more indicative naming for these variables :+1:

Reply:
I went ahead updated all occurrences of `x` as a variable here, and introduced more indicative naming 👍